manifests/charon.pp

```diff
@@ -1,44 +1,39 @@
-# Define strongswan::logging
-# ===========================
+# @summary Manage strongswan charon config files
 #
-# Manage strongswan charon.conf
+# @param options
+#   charon options hash
 #
-# Example configuration:
-# ===========================
-#
-# strongswan::charon { 'log':
-#   options => {
-#     'filelog' => {
-#       '/var/log/strongswan.log' => {
-#         'time_format' => '%b %e %T',
-#         'ike_name'    => 'yes',
-#         'append'      => 'no',
-#         'default'     => '1',
-#         'flush_line'  => 'yes',
+# @example Define file loggers in charon_log.conf file
+#   strongswan::charon { 'log':
+#     options => {
+#       'filelog' => {
+#         '/var/log/strongswan.log' => {
+#           'time_format' => '%b %e %T',
+#           'ike_name'    => 'yes',
+#           'append'      => 'no',
+#           'default'     => '1',
+#           'flush_line'  => 'yes',
+#         },
+#         'stderr' => {
+#           'ike' => '2',
+#           'knl' => '2',
+#         },
 #       },
-#       'stderr' => {
-#         'ike' => '2',
-#         'knl' => '2',
-#       }
-#     }
+#     },
 #   }
-# }
-#
-
 define strongswan::charon (
-  $options    = {},
+  Hash $options = {},
 ) {
+  require strongswan::package
```

Comment:
which includes `strongswan`.  We make sure both the variables we need below have been resolved and that the configuration directory will have been created by the package.